model_factory.py

```diff
@@ -0,0 +1,23 @@
+from enum import Enum, unique
+from models.simple_embedding_model import SimpleEmbeddingModel
+from models.bert_feature_extractor_model import BERTAsFeatureExtractorEncoder, BERTVersion
+
+
+@unique
+class ModelType(Enum):
+    BERT_AVG_LAYER_TOKEN_FEATURE_EXTRACTOR = "bert_avg_layer_token_feature_extractor"
+    SIMPLE_EMBEDDING_MODEL = "simple_embedding_model"
+
+
+def get_model(model_type: ModelType, train_config):
```

Comment:
Nitpick: something like below should require less boilerplace:

```python
class ModelFactory:
    @staticmethod
    def bert_avg_layer_token_feature_extractor(train_config):
        return BERTAsFeatureExtractorEncoder(
            BERTVersion.BASE_UNCASED,
            hidden_size=train_config["embedding_dim"]
        )

    @staticmethod
    def simple_embedding_model(train_config):
        return SimpleEmbeddingModel(train_config["embedding_dim"])

model = getattr(ModelFactory, args.model_type)(vars(args))

```

Reply:
I went ahead and committed that proposal, feel free to revert if you don't like it @ItsWayneYang